src/backend/joanie/core/models/courses.py

```diff
@@ -914,8 +914,10 @@ def clean(self):
         `was_created_by_order` cannot be set to True.
         """
 
-        # The related course run must be opened for enrollment
-        if self.course_run.state["priority"] > CourseState.ARCHIVED_OPEN:
+        # The related course run must be opened to create the enrollment or reactivate it.
+        if (not self.created_on or self.is_active) and self.course_run.state[
+            "priority"
+        ] > CourseState.ARCHIVED_OPEN:
```

Comment:
FMI: 
Why are we adding `not self.created_on` here ? should it be tested, or is it already with an existing test ?
I don't really get the `>`. I see that course states values are build with a range, but I don't get the logic behind `>`.

Reply:
## About `self.created_on` condition statement

From my point of view it is relevant to prevent to create an enrollment even if it is inactive when the course run is closed. [There is a test about that](https://github.com/openfun/joanie/blob/a50bf6162324fcaa3cebf07e04879f6f2703aa71/src/backend/joanie/tests/core/test_models_enrollment.py#L364-L380).


## About `>` operator.

`CourseState` has class attributes which describe each allowed priority values (From 0 to 7). Then when we processed the state of a course run, we can retrieve its priority through course_run.state['priority'].

[The trick here is the order of priority values](https://github.com/openfun/joanie/blob/a50bf6162324fcaa3cebf07e04879f6f2703aa71/src/backend/joanie/core/models/courses.py#L42-L51). Indeed more the priority is high more the course run is "ended". So now here it is the explanation of the condition statement:
We are testing that course_run has a priority greater than 2 (`ARCHIVED_OPEN`). If you check this CourseState priority class attributes, you will see that values greater than 2 represents only course run closed for enrollment (`FUTURE_NOT_YET_OPEN`, `FUTURE_CLOSED`, `ONGOING_CLOSED`, `ARCHIVED_CLOSED`, `TO_BE_SCHEDULED`).

To resume `course_run.state['priority'] > 2` is an easy check to ensure that course run is opened for enrollment.

Reply:
Oh, thanks, I didn't see the docstring :+1: